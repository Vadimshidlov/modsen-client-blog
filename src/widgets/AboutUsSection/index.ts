export { AboutUsSection } from '@/widgets/AboutUsSection/ui/AboutUsSection';
