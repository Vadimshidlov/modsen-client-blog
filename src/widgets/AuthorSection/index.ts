export { AuthorSection } from '@/widgets/AuthorSection/ui/AuthorSection';
