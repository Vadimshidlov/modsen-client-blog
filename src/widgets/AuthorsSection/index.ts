export { AuthorsSection } from '@/widgets/AuthorsSection/ui/AuthorsSection';
