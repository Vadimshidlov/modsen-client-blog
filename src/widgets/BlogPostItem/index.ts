export { BlogPostItem } from '@/widgets/BlogPostItem/BlogPostItem';
