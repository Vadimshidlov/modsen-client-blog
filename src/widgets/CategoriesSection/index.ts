export { CategoriesSection } from '@/widgets/CategoriesSection/ui/CategoriesSection';
