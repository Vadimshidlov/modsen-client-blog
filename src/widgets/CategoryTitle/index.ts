export { CategoryTitle } from '@/widgets/CategoryTitle/ui/CategoryTitle';
