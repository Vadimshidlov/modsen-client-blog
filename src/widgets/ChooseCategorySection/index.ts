export { ChooseCategorySection } from '@/widgets/ChooseCategorySection/ui/ChooseCategorySection';
