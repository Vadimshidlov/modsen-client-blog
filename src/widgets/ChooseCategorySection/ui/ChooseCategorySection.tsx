import React from 'react';
import styles from '@/widgets/ChooseCategorySection/ui/ChooseCategorySection.module.scss';
import { CATEGORIES } from '@/shared/constants/categories';
import { Category } from '@/widgets/ChooseCategorySection/ui/Category';
import container from '@/shared/styles/container.module.scss';
import { useTranslations } from 'next-intl';

export function ChooseCategorySection() {
  const t = useTranslations('ChooseCategorySection');

  return (
    <section className={`${styles.chooseCategoryContainer} ${container.container}`}>
      <h3 className={styles.sectionTitle}>{t('title')}</h3>
      <ul className={styles.categoriesContainer}>
        {CATEGORIES.map(({ img, title, text }) => (
          <Category img={img} text={text} title={title} key={title} />
        ))}
      </ul>
    </section>
  );
}
