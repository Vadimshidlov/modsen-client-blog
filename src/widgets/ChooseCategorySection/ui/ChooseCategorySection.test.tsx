import { render } from '@testing-library/react';
import { ChooseCategorySection } from '@/widgets/ChooseCategorySection';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('ChooseCategorySection tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<ChooseCategorySection />);
    expect(container).toMatchSnapshot();
  });
});
