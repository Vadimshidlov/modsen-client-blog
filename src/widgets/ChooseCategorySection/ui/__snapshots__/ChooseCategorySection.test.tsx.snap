// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseCategorySection tests should matches snapshot 1`] = `
<div>
  <section
    class="chooseCategoryContainer container"
  >
    <h3
      class="sectionTitle"
    >
      Mocked title
    </h3>
    <ul
      class="categoriesContainer"
    >
      <li
        class="categoriesItem"
      >
        <a
          data-testid="category-link-business"
          href="/category/business"
        >
          <img
            alt="category-image"
            class="categoryImage"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <h4
            class="categoryTitle"
          >
            Mocked Business
          </h4>
          <span
            class="categoryText"
          >
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
          </span>
        </a>
      </li>
      <li
        class="categoriesItem"
      >
        <a
          data-testid="category-link-startup"
          href="/category/startup"
        >
          <img
            alt="category-image"
            class="categoryImage"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <h4
            class="categoryTitle"
          >
            Mocked Startup
          </h4>
          <span
            class="categoryText"
          >
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
          </span>
        </a>
      </li>
      <li
        class="categoriesItem"
      >
        <a
          data-testid="category-link-economy"
          href="/category/economy"
        >
          <img
            alt="category-image"
            class="categoryImage"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <h4
            class="categoryTitle"
          >
            Mocked Economy
          </h4>
          <span
            class="categoryText"
          >
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
          </span>
        </a>
      </li>
      <li
        class="categoriesItem"
      >
        <a
          data-testid="category-link-technology"
          href="/category/technology"
        >
          <img
            alt="category-image"
            class="categoryImage"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <h4
            class="categoryTitle"
          >
            Mocked Technology
          </h4>
          <span
            class="categoryText"
          >
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
          </span>
        </a>
      </li>
    </ul>
  </section>
</div>
`;
